tests/scripts/all.sh

```diff
@@ -2633,6 +2629,235 @@ component_test_psa_crypto_config_reference_ecc_no_ecp_at_all () {
+common_tfm_config () {
+    # Enable TF-M config
+    cp configs/tfm_mbedcrypto_config_profile_medium.h "$CONFIG_H"
+    cp configs/crypto_config_profile_medium.h "$CRYPTO_CONFIG_H"
+
+    # Adjust for the fact that we're building outside the TF-M environment.
+    #
+    # TF-M has separation, our build doesn't
+    scripts/config.py unset MBEDTLS_PSA_CRYPTO_SPM
+    scripts/config.py unset MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
+    # TF-M provdes its own (dummy) implemenation, from their tree
+    scripts/config.py unset MBEDTLS_AES_DECRYPT_ALT
+    scripts/config.py unset MBEDTLS_AES_SETKEY_DEC_ALT
+    # We have an OS that provides entropy, use it
+    scripts/config.py unset MBEDTLS_NO_PLATFORM_ENTROPY
+
+    # Other config adjustments to make the tests pass.
+    # Those are a surprise and should be investigated and fixed.
+    #
+    # pkparse.c and pkwrite.c fail to link without this
+    echo "#define MBEDTLS_OID_C" >> "$CONFIG_H"
+
+    # Config adjustements for better test coverage in our environment.
+    # These are not needed just to build and pass tests.
+    #
+    # Enable filesystem I/O for the benefit of PK parse/write tests.
+    echo "#define MBEDTLS_FS_IO" >> "$CONFIG_H"
+}
+
+# Keep this in sync with component_test_tfm_config() as they are both meant
+# to be used in analyze_outcomes.py for driver's coverage analysis.
+component_test_tfm_config_p256m_driver_accel_ec () {
+    msg "build: TF-M config + p256m driver + accel ECDH(E)/ECDSA"
+
+    common_tfm_config
+
+    # Disable all the features that auto-enable ECP_LIGHT (see build_info.h)
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE
+
+    # Add missing symbols from "tfm_mbedcrypto_config_profile_medium.h"
+    #
+    # - USE_PSA_CRYPTO for PK_HAVE_ECC_KEYS
+    echo "#define MBEDTLS_USE_PSA_CRYPTO" >> "$CONFIG_H"
+    # - ECP_C and BIGNUM because P256M does not have support for import and export
+    #       of keys so we need the builtin support for that
+    echo "#define MBEDTLS_ECP_C" >> "$CONFIG_H"
+    echo "#define MBEDTLS_BIGNUM_C" >> "$CONFIG_H"
+    # - ASN1_[PARSE/WRITE]_C and OID_C found by check_config.h
+    echo "#define MBEDTLS_ASN1_PARSE_C" >> "$CONFIG_H"
+    echo "#define MBEDTLS_ASN1_WRITE_C" >> "$CONFIG_H"
+    echo "#define MBEDTLS_OID_C" >> "$CONFIG_H"
+
+    # Set the list of accelerated components in order to remove them from
+    # builtin support. We don't set IMPORT and EXPORT because P256M does not
+    # support these operations.
+    loc_accel_list="ALG_ECDSA \
+                    ALG_ECDH \
+                    KEY_TYPE_ECC_KEY_PAIR_BASIC \
+                    KEY_TYPE_ECC_KEY_PAIR_GENERATE \
+                    KEY_TYPE_ECC_PUBLIC_KEY"
+    loc_accel_flags="$( echo "$loc_accel_list" | sed 's/[^ ]* */-DMBEDTLS_PSA_ACCEL_&/g' )"
+
+    # Build crypto library specifying we want to use P256M code for EC operations
+    make CFLAGS="$loc_accel_flags -DMBEDTLS_P256M_EXAMPLE_DRIVER_ENABLED"
```

Comment:
Note: I feel we should build with ASan here (like we do for other driver builds).

Reply:
I tried to add `ASAN_CFLAGS` to the build, but there seems to be a runtime error related to the `-fsanitize=undefined` option:
```
memory_buffer_alloc.c:311:15: runtime error: member access within misaligned address 0x7ffdcab0144c for type 'struct memory_header', which requires 8 byte alignment
```

Reply:
Ah right, so there's a compromise here, we can either:
1. test with ASan and disable `BUFFER_ALLOC_C` for maximum ASan efficiency, at the expense of similarity with the TF-M config;
2. test with ASan and `BUFFER_ALLOC_C` for maximum similarity with TF-M config, at the expense of ASan efficiency (it would still check stack buffers and static ones, and we'd still have UBSan enabled); 
3. keep `BUFFER_ALLOC_C` and don't bother with ASan.

I don't think (3) has much merit, and I'd actually be inclined to go for (1). IMO we want to test a config that's identical to TF-M's in terms of crypto features, but we don't care as much about platform-type things - we are already modifying some of them anyway. But I'm open to argument for (2).

Reply:
Side note, since I faced this problem yesteday when I  tried to address this comment before @gilles-peskine-arm's suggestion. In case we go for option (2) the `MBEDTLS_MEMORY_ALIGN_MULTIPLE` should be set to 8 bytes in 64 bit architectures otherwise tests fail due to wrong alignment.

Reply:
I think we want maximum ASan efficiency for these tests

Reply:
Marking as resolved as this has been done: https://github.com/Mbed-TLS/mbedtls/pull/8041/files#diff-e683396c93d3af1ea2db5a3d94788da3916ed084f3eb8be5e5f7c0e429badf39R2820